File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        
        int currMax = nums[0],globalMax = nums[0];
        for(int i=1;i<nums.size();i++)
        {
            currMax = max(nums[i],currMax+nums[i]);
            globalMax = max(currMax,globalMax);
        }
        return globalMax;

    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) 
    {
        int gmax=nums[0],lmax=nums[0];
        for(int i=1;i<nums.length;i++)
        {
            lmax = Math.max(nums[i],lmax+nums[i]);
            gmax = Math.max(lmax,gmax);
        }
        return gmax;
    }
}